solutions/standard/main.tf

```diff
@@ -0,0 +1,253 @@
+          # be included if all indiviual values are set to null
+          : length([
+            for value in ["port_min", "port_max", "source_port_min", "source_port_min"] :
+            true if lookup(rules.value["tcp"], value, null) == null
+          ]) == 4
+          ? []
+          : [rules.value]
+        )
+        content {
+          port_min        = lookup(rules.value.tcp, "port_min", null)
+          port_max        = lookup(rules.value.tcp, "port_max", null)
+          source_port_min = lookup(rules.value.tcp, "source_port_min", null)
+          source_port_max = lookup(rules.value.tcp, "source_port_max", null)
+        }
+      }
+
+      dynamic "udp" {
+        for_each = (
+          # if rules null
+          rules.value.udp == null
+          # empty array
+          ? []
+          # otherwise check each possible field against how many of the values are
+          # equal to null and only include rules where one of the values is not null
+          # this allows for patterns to include `udp` blocks for conversion to list
+          # while still not creating a rule. default behavior would force the rule to
+          # be included if all indiviual values are set to null
+          : length([
+            for value in ["port_min", "port_max", "source_port_min", "source_port_min"] :
+            true if lookup(rules.value["udp"], value, null) == null
+          ]) == 4
+          ? []
+          : [rules.value]
+        )
+        content {
+          port_min        = lookup(rules.value.udp, "port_min", null)
+          port_max        = lookup(rules.value.udp, "port_max", null)
+          source_port_min = lookup(rules.value.udp, "source_port_min", null)
+          source_port_max = lookup(rules.value.udp, "source_port_max", null)
+        }
+      }
+
+      dynamic "icmp" {
+        for_each = (
+          # if rules null
+          rules.value.icmp == null
+          # empty array
+          ? []
+          # otherwise check each possible field against how many of the values are
+          # equal to null and only include rules where one of the values is not null
+          # this allows for patterns to include `udp` blocks for conversion to list
+          # while still not creating a rule. default behavior would force the rule to
+          # be included if all indiviual values are set to null
+          : length([
+            for value in ["code", "type"] :
+            true if lookup(rules.value["icmp"], value, null) == null
+          ]) == 2
+          ? []
+          : [rules.value]
+        )
+        content {
+          type = rules.value.icmp.type
+          code = rules.value.icmp.code
+        }
+      }
+    }
+  }
+}
+
+# input: restrict to the clients IPS () - required - maybe in descriptio u can cfeate 0.0.0 to use as it oepn
+
+resource "ibm_is_subnet" "client_to_site_subnet_zone_1" {
+  depends_on      = [ibm_is_vpc_address_prefix.client_to_site_address_prefixes_zone_1]
```

Comment:
suggest to use `count = 1` to future proof for existing subnet support (do this for all applicable resources)